chapters/annotations.tex

```diff
@@ -8,6 +8,26 @@ \chapter{Annotations}\label{annotations}
 (For replaceable class declarations with a \lstinline[language=grammar]!constraining-clause! also refer to \cref{constraining-clause-annotations}.)
 The specification in this document defines the semantic meaning if a tool implements any of these annotations.
 
+\section{Notation for Annotation Definitions}\label{notation-for-annotation-definitions}
+
+Annotations are defined using the syntactic forms of Modelica component declarations and record definitions, with the following special semantics.
+If the annotation is described by a component declaration, the annotation is used in the form of a value modifier for the same name.
+If the annotation is described by a \lstinline!record! \emph{class} the annotation is used in the form of a modifier for a \lstinline!record! \emph{component} with the same name.
```

Comment:
Now that we are handling annotation as some sort of modifiers, I think we need to clarify this difference:
```
annotation(Dialog(loadSelector)); /* No effect at all */
annotation(Dialog(loadSelector = Selector())); /* In effect, using defaults for Selector. */
```